### DISCUSSION TITLE
富文本支持

### DISCUSSION THREAD

**Discussion by peter100u** [Tencent/libpag#376] [Q&A] [answered]
我看demo中仅仅是简单文本替换和样式的修改，没看到富文本的API

**Comment by domchen** [comment, marked as answer]
你好，PAG目前还没有实现富文本的支持，AE中的富文本目前也是导出不了的，所以对应的没有相关编辑API。可以描述一下具体是什么场景下用到的富文本吗？过往的场景中也比较少遇见相关的需求。我们看一下你具体的使用场景是否有更优的解决思路或者增加一下这块的排期。但富文本相关的内容整体的工作量还是非常大的。时间上需要有预期。

**Reply by peter100u** [reply]
场景是在一个语音房中，高等级用户进来的时候需要播放一个进场特效，特效中有文件描述如下：
欢迎Peter进入大富豪的房间， （Peter这个用户名需要黄色加粗展示（并且支持点击peter这个用户名，进入他的个人资料页面）， 大富豪这个房间名需要黄色加租展示）

**Reply by domchen** [reply]
整体的文本需要有动画是吧？其实目前可以不用等待富文本直接就实现。思路是用系统的富文本能力，排版完了截个图出来，PAG那边就只管动画的部分，放个占位图在那边附带预设动画效果，运行时你把富文本的截图替换进去就行了。但点击的部分实现起来可能还是有点麻烦。整体上还是建议拆分开做，不用都放在PAG里实现。这部分文本应该也是编程控制的动画所以效果并不复杂，直接用代码实现了盖在PAG其他动画的上层应该是最简单的。

**Reply by peter100u** [reply]
是的呢，目前就是代码实现的，但是以后文字特效复杂了，用原生的andorid shader还是很费劲

**Reply by domchen** [reply]
了解，目前先用分开的做法吧。我们也研究一下富文本这块的支持。
